### ISSUE TITLE
comment_references: Analyzer does not always detect member references in documentation

### ISSUE THREAD

**Issue by bergwerf** [dart-lang/linter#1878]
It appears the regex for matching the `variable` in `/// This function uses [variable].` is a bit too restrictive. If I add a character like `,` or `:` directly after the `]` (for example `/// + [arg1]: Use to do ...`), then it is no longer recognized (I rely on the analyzer to tell me if I reference out of scope things in my comments).

**Comment by srawlins** [comment]
I can reproduce this. Modify line 13 from `test/rules/comment_references.dart`:

```dart
/// Writes [y]: yay. #LINT                                                                                                                             
void write(int x) {}
```

and no lint is raised :(

**Comment by bwilkerson** [comment]
I suspect that this is because of an attempt to support `[id]: http://example.com/` style link labels.

The underlying difficulty is that doc comments are an extended form of markdown in which square brackets are used both for standard markdown links and for scope-based links.

**Comment by srawlins** [comment]
Fixed with https://github.com/dart-lang/sdk/commit/96070e3b5d15f1dfb402ebc50b59faecfbf88603
